src/extensions/extensions-store.ts

```diff
@@ -0,0 +1,49 @@
+import type { LensExtensionId } from "./lens-extension";
+import { BaseStore } from "../common/base-store"
+import { action, observable, toJS } from "mobx";
+
+export interface LensExtensionsStoreModel {
+  extensions: Record<LensExtensionId, LensExtensionState>;
+}
+
+export interface LensExtensionState {
+  enabled?: boolean;
+}
+
+export class ExtensionsStore extends BaseStore<LensExtensionsStoreModel> {
+  constructor() {
+    super({
+      configName: "lens-extensions"
+    });
+  }
+
+  @observable extensions = observable.map<LensExtensionId, LensExtensionState>();
+
+  @action
+  setEnabled(extId: LensExtensionId, enabled: boolean) {
+    const state = this.extensions.get(extId);
+    this.extensions.set(extId, {
+      ...(state || {}),
+      enabled: enabled,
+    })
+  }
+
+  isEnabled(extensionId: LensExtensionId) {
+    const state = this.extensions.get(extensionId);
+    return !state /* enabled by default */ || state.enabled;
```

Comment:
I think that is would be clearer if the following was used. Because the null-coelessing operator is exactly what we are trying to model.

```suggestion
    return state?.enabled ?? true
```

Reply:
I personally don't like `??` operator and `||` looks more logically as `OR`.
So if checking something that expected to be an object I prefer `||`.

Reply:
Is it just how the `??` operator looks? Because from a meaning standpoint `??` is closer to what you want than `||`.